modules/udf/main.tf

```diff
@@ -14,113 +14,29 @@
-      WHEN STRPOS(str, ',') > 0 THEN 0
-      ELSE
-      (
-        WITH list AS (
-          SELECT ROW_NUMBER() OVER() id, l FROM UNNEST(SPLIT(strList, ',')) l
-        )
-        (SELECT id FROM list WHERE l = str)
-      )
-    END
-  );"
-EOT
-
-  destroy_cmd_body = "--project_id ${var.project_id} query --use_legacy_sql=false \"DROP FUNCTION IF EXISTS ${var.dataset_id}.find_in_set\""
-}
-
-module "bq_check_protocol" {
-  source  = "github.com/terraform-google-modules/terraform-google-gcloud"
-  enabled = var.add_udfs
-
-  platform              = "linux"
-  additional_components = ["bq"]
-
-  create_cmd_entrypoint  = "bq"
-  destroy_cmd_entrypoint = "bq"
-
-  create_cmd_body = <<EOT
---project_id ${var.project_id} query --use_legacy_sql=false "CREATE FUNCTION IF NOT EXISTS ${var.dataset_id}.check_protocol(url STRING)
-  AS (
-    CASE
-      WHEN REGEXP_CONTAINS(url, '^[a-zA-Z]+://') THEN url
-      ELSE CONCAT('http://', url)
-    END
-  );"
-EOT
-
-  destroy_cmd_body = "--project_id ${var.project_id} query --use_legacy_sql=false \"DROP FUNCTION IF EXISTS ${var.dataset_id}.check_protocol\""
-}
-
-module "bq_parse_url" {
-  source  = "github.com/terraform-google-modules/terraform-google-gcloud"
-  enabled = module.bq_check_protocol.wait != "" && var.add_udfs
-
-  platform              = "linux"
-  additional_components = ["bq"]
-
-  create_cmd_entrypoint  = "bq"
-  destroy_cmd_entrypoint = "bq"
-
-  create_cmd_body = <<EOT
---project_id ${var.project_id} query --use_legacy_sql=false "CREATE FUNCTION IF NOT EXISTS ${var.dataset_id}.parse_url(url STRING, part STRING)
-  AS (
-    CASE
-      -- Return HOST part of the URL.
-      WHEN part = 'HOST' THEN SPLIT(\`${var.project_id}\`.${var.dataset_id}.check_protocol(url), '/')[OFFSET(2)]
-      WHEN part = 'REF' THEN
-        IF(REGEXP_CONTAINS(url, '#'), SPLIT(\`${var.project_id}\`.${var.dataset_id}.check_protocol
-        (url), '#')[OFFSET(1)], NULL)
-      WHEN part = 'PROTOCOL' THEN RTRIM(REGEXP_EXTRACT(url, '^[a-zA-Z]+://'), '://')
-      ELSE ''
-    END
-  );"
-EOT
-
-  destroy_cmd_body = "--project_id ${var.project_id} query --use_legacy_sql=false \"DROP FUNCTION IF EXISTS ${var.dataset_id}.parse_url\""
+locals {
+  // Regex to extract function name (optionally might include project and dataset ids)
+  // see https://cloud.google.com/bigquery/docs/reference/standard-sql/user-defined-functions#temporary-udf-syntax
+  // You can view test cases at: https://regex101.com/r/a7CNA6/2/
+  function_name_regex = "CREATE\\s+(?:OR REPLACE\\s+)?FUNCTION\\s+(?:IF NOT EXISTS)?\\s*([a-zA-Z0-9_]+)(?=\\s*\\()"
+  function_map = {
+    for udf_query in var.udf_ddl_queries :
+    regex(local.function_name_regex, udf_ddl_query) => udf_query
+  }
 }
 
-module "bq_csv_to_struct" {
-  source  = "github.com/terraform-google-modules/terraform-google-gcloud"
-  enabled = var.add_udfs
+module "udf" {
+  for_each = local.function_map
+  source   = "github.com/terraform-google-modules/terraform-google-gcloud"
```

Comment:
terraform-google-modules/gcloud/google

Also please move the `source` field to be the first field and also add a version.